src/udev/udev-ctrl.c

```diff
@@ -180,6 +180,11 @@ static int udev_ctrl_connection_event_handler(sd_event_source *s, int fd, uint32
                 return 0;
         if (size < 0)
                 return log_error_errno(size, "Failed to receive ctrl message: %m");
+        if ((size_t) size != sizeof(UdevCtrlMessageWire)) {
```

Comment:
size would exceed iov.len only if MSG_TRUNC is specified as flag for the recvmsg call. A reliable way to check for too long packets is FLAGS_SET(msg->msg_flags, MSG_TRUNC)